schema/crdb/4.0.0/up.sql

```diff
@@ -0,0 +1,25 @@
+-- CRDB documentation recommends the following:
+--  "Execute schema changes either as single statements (as an implicit transaction),
+--   or in an explicit transaction consisting of the single schema change statement."
+--
+-- For each schema change, we transactionally:
+-- 1. Check the current version
+-- 2. Apply the idempotent update
+
+BEGIN;
+
+SELECT CAST(
+    IF(
+        (
+            SELECT version = '3.0.3' and target_version = '4.0.0'
+            FROM omicron.public.db_metadata WHERE singleton = true
+        ),
+        'true',
+        'Invalid starting version for schema change'
+    ) AS BOOL
+);
+
+ALTER TYPE omicron.public.service_kind ADD VALUE IF NOT EXISTS 'clickhouse_keeper';
+ALTER TYPE omicron.public.dataset_kind ADD VALUE IF NOT EXISTS 'clickhouse_keeper';
+
+COMMIT;
```

Comment:
I know this might seem like a nitpick, but IMO there is an important property described in the schema changing README should respect here: 

https://github.com/oxidecomputer/omicron/blob/01e730acf3efeb6ff2181d558cf373ded383a365/schema/crdb/README.adoc?plain=1#L50

There's a lot more background in https://www.cockroachlabs.com/docs/stable/online-schema-changes#schema-changes-within-transactions , but basically: CRDB docs really strongly recommend against this "multi-statement alteration" within a transaction. In this code, both `ALTER TYPE` calls are statements, so we have "two per transaction".

In this **particular** case, I doubt it would bite us, but I think we should be cautious as a precedent, while we can.

One really easy workaround: we can just issue two transactions from this file, since each one should be idempotent and independent:

```sql
BEGIN;
SELECT CAST ( # Check version...
ALTER TYPE omicron.public.service_kind ADD VALUE IF NOT EXISTS 'clickhouse_keeper';
COMMIT;

BEGIN;
SELECT CAST ( # Check version, again
ALTER TYPE omicron.public.dataset_kind ADD VALUE IF NOT EXISTS 'clickhouse_keeper';
COMMIT;
```

Reply:
Sorry! tbh I missed that bit. I can do it the correct way, I've added an additional file :)